### PR TITLE
V10 height trigger

### PR DESCRIPTION
**Pull request by denravonska**
Changes v10 initiation to use a height trigger instead of based on the version of the previous block.

### CODE REVIEW

**Comment by denravonska** [comment]
The v10 switch in prod is set to an even number a bit less than 1 month from now.

File: src/miner.cpp [hunk 1]
```diff
@@ -1049,7 +1049,7 @@ void StakeMiner(CWallet *pwallet)
                 StakeBlock.nVersion = 8;
             if(IsV9Enabled(pindexPrev->nHeight+1))
                 StakeBlock.nVersion = 9;
-            if(pindexPrev->nVersion == 10)
+            if(IsV10Enabled(pindexPrev->nHeight + 1))
                 StakeBlock.nVersion = 10;
```
Comment (tomasbrod) [inline src/miner.cpp:1053]: It might be worth to remove the other alternatives. Wont be mining bv8 blocks anymore.

**Review by tomasbrod — APPROVED** [review]
Checked: OK.

**Comment by tomasbrod** [comment]
Note that v3.7.16.2 *is* forward compatible with this change.

**Pull request merged**